stream/src/public_key/connection.rs

```diff
@@ -125,6 +125,11 @@ impl<Si: Sink, St: Stream> Connection<Si, St> {
         mut stream: St,
         peer: C::PublicKey,
     ) -> Result<Self, Error> {
+        // Ensure we are not trying to connect to ourselves
+        if peer == config.crypto.public_key() {
```

Comment:
I "think" we could get away with asserting here. We take that approach for things that are indicative of a bug (something that should never be attempted in `storage`).

Reply:
I don't think we really do that in `p2p` or `stream`, so ok leaving as is for consistency.